clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DeployAtomicOperation.groovy

```diff
@@ -54,12 +54,13 @@ class DeployAtomicOperation implements AtomicOperation<DeploymentResult> {
       throw new DeployHandlerNotFoundException("Could not find handler for ${description.getClass().simpleName}!")
     }
 
-    deployHandler.preDeploy(this);
     task.updateStatus TASK_PHASE, "Found handler: ${deployHandler.getClass().simpleName}"
 
     task.updateStatus TASK_PHASE, "Invoking Handler."
     def deploymentResult = deployHandler.handle(description, priorOutputs)
 
+    events.addAll(deployHandler.getDeployEvents() ?: [])
```

Comment:
no. this was to fix a failing test. But I can mock the deploy.getEvents() to return an empty list.